### PR TITLE
Implement nanquantile function in keras.ops

### PR DESCRIPTION
**Pull request by shashaka**
Adds keras.ops.nanquantile, which computes the q-th quantile of the input tensor along a specified axis while ignoring NaN values.
Supported across NumPy, TensorFlow, PyTorch, and JAX backends. Not supported on OpenVINO.

### CODE REVIEW

File: keras/src/backend/torch/numpy.py [hunk 1]
```diff
@@ -1430,6 +1430,45 @@ def nanprod(x, axis=None, keepdims=False):
     )
 
 
+def nanquantile(x, q, axis=None, method="linear", keepdims=False):
+    x = convert_to_tensor(x)
+    q = convert_to_tensor(q)
+    axis = to_tuple_or_list(axis)
+
+    compute_dtype = dtypes.result_type(x.dtype, "float32")
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/torch/numpy.py:1448]: The line is intentional. If it is removed, the dtype tests fail for non-floating inputs

File: keras/src/backend/tensorflow/numpy.py [hunk 2]
```diff
@@ -2308,6 +2308,72 @@ def nanprod(x, axis=None, keepdims=False):
     return prod(x_safe, axis=axis, keepdims=keepdims)
 
 
+def nanquantile(x, q, axis=None, method="linear", keepdims=False):
+    x = convert_to_tensor(x)
+    q = convert_to_tensor(q, dtype=x.dtype)
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/tensorflow/numpy.py]: Updated.
